cluster-autoscaler/scale_down.go

```diff
@@ -163,6 +164,31 @@ func ScaleDown(
 		return ScaleDownNoUnneeded, nil
 	}
 
+	emptyNodes := filterEmptyNodes(simulator.FindEmptyNodesToRemove(candidates, pods), maxEmptyBulkDelete, cloudProvider)
```

Comment:
Can you please add a comment that will explain that in this section you try to delete empty nodes in bulks, and if this fails then you will try to delete 1 non-empty node? A comment in this next section would also help.